http-protocol-binding.md

```diff
@@ -429,6 +429,57 @@ Content-Length: nnnn
 
 ```
 
+### 3.4. Multipart Binary Content Mode
+
+In the _multipart binary_ content mode several events are sent into a single HTTP multipart
+request or response body and they can be named. A multipart binary mode contains one event per part, 
+encoded in a similar fashion to [binary content mode]().
+The _multipart binary_ content mode is based on [RFC2046][rfc2046].
+
+#### 3.4.1. HTTP Content-Type
+
+The [HTTP `Content-Type`][content-type] header MUST be set to the media type of `multipart/cloudevents` 
+and must include the [boundary parameter](https://tools.ietf.org/html/rfc2046#section-5.1.1)
+
+```text
+Content-Type: multipart/cloudevents; boundary=12345
+```
+
+#### 3.4.2 Examples
+
+```text
+
+POST /myresource HTTP/1.1
+Host: webhook.example.com
+Content-Type: multipart/cloudevents; boundary=12345
+
+--12345
+ce-specversion: 1.0
```

Comment:
since we're talking about a binary content mode, it would be nice to reduce the overhead from headers.

Otherwise, header prefixes may take more than 50% of the part size and the format may become ineffective.

Reply:
Maybe the top header can include some sort of a table that assigns indexes to (common) headers. Something like:
```http
POST /myresource HTTP/1.1
Host: webhook.example.com
Content-Type: multipart/cloudevents; boundary=12345
ce-hpack: specversion=0&type=1&time=2&id=3&source=4 /* etc etc */

--12345
0: 1.0
1: com.example.someevent
2: 2018-04-05T03:56:24Z
3: 1234-1234-1234
4: /mycontext/subcontext
```

(not a final proposal, just to demonstrate the idea)

Reply:
Probably we could define the `ce-hpack` thing directly from the spec

Reply:
Now that I look again at this, I'm not sure this should be done by this spec directly, because HTTP/2 already has the header compression.

Reply:
does it do it for multipart body tho?

Reply:
I'm not sure, but I don't think so...